src/components/DownloadTrajectoryMenu/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import { Button, Tooltip } from "antd";
+import { ISimulariumFile } from "@aics/simularium-viewer/type-declarations";
+
+import { DATA_BUCKET_URL, TOOLTIP_COLOR } from "../../constants";
+import {
+    NetworkedSimFile,
+    LocalSimFile,
+    isNetworkSimFileInterface,
+} from "../../state/trajectory/types";
+import { Download } from "../Icons";
+
+import styles from "./style.css";
+
+interface DownloadTrajectoryMenuProps {
+    isBuffering: boolean;
+    simulariumFile: LocalSimFile | NetworkedSimFile;
+}
+
+const DownloadTrajectoryMenu = ({
+    isBuffering,
+    simulariumFile,
+}: DownloadTrajectoryMenuProps): JSX.Element => {
+    const fileIsLoaded = () => !!simulariumFile.name;
+
+    const getHref = () => {
+        if (!fileIsLoaded()) {
+            return "";
+        }
+        if (isNetworkSimFileInterface(simulariumFile)) {
+            return `${DATA_BUCKET_URL}/trajectory/${simulariumFile.name}`;
+        } else {
+            const data: ISimulariumFile = simulariumFile.data;
+            const blob = data.getAsBlob();
+            return URL.createObjectURL(blob);
+        }
+    };
+
+    const downloadFile = async (fileName: string): Promise<void> => {
+        const downloadLink = document.createElement("a");
+        downloadLink.download = fileName;
+        downloadLink.style.display = "none";
+        downloadLink.href = getHref();
+        document.body.appendChild(downloadLink);
+        downloadLink.click();
+        document.body.removeChild(downloadLink);
+        URL.revokeObjectURL(downloadLink.href);
+    };
```

Comment:
This function doesn't need to be `async` unless it contains an `await` (I'm pretty sure).

Reply:
You are correct